source/agora/node/Validator.d

```diff
@@ -533,11 +533,15 @@ public class Validator : FullNode, API
 
         Params:
             block_height = Current block height
+            dont_wait = Immediately send the request.
 
     ***************************************************************************/
 
-    protected void checkAndEnroll (Height block_height) @safe
+    protected void checkAndEnroll (Height block_height, bool dont_wait = false) @safe
     {
+        // Enrollment emitter
+        static ITimer emitter;
```

Comment:
`static` variables within member functions are evil. Usually the unit of encapsulation is the object, but in this case it's the method, meaning that if we ever have 2 nodes running in the same thread, things will start to fail randomly. I'd recommend to steer clear from them.